src/vdom/VNode.ts

```diff
@@ -141,25 +144,51 @@ export class VNode implements Comparable<VNode> {
     }
 
     public getInnerHtml(): string {
+        const locale = getLocale();
+        const htmlToUse = this.rawInnerHtml != undefined ? this.rawInnerHtml : this.innerHtml;
         if (this.app.i18nResolver != undefined
-            && this.app.i18nResolver.some(resolver => this.innerHtml.startsWith(resolver.getPrefix()))) {
-            const resolver = this.app.i18nResolver.filter(resolver => this.innerHtml.startsWith(resolver.getPrefix()));
-            let localized: string = undefined;
-            for(const res of resolver) {
-                localized = res.get(this.innerHtml, getLocale())
-                if(localized != undefined) {
-                    break;
-                }
-            }
-
-            if (localized != undefined) {
-                return new Stringparser().parse(localized, this.props);
-            }
+            && this.app.i18nResolver.some(resolver => htmlToUse.startsWith(resolver.getPrefix()))
+            && locale != this.lastResolvedLocale)
+        {
+            this.resolvei18n().catch((e) => console.error(e));
         }
-
         return new Stringparser().parse(this.innerHtml, this.props);
     }
 
+    private resolvei18n(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            const locale = getLocale();
+            const htmlToUse = this.rawInnerHtml != undefined ? this.rawInnerHtml : this.innerHtml;
+            // Figure out relevant resolver and map their get method into always-resolving promises
+            const resolver = this.app.i18nResolver.sort((a,b) => {
+                if (a.isStrict && !b.isStrict()) return -1;
+                if (!a.isStrict && b.isStrict()) return 1;
+                return 0;
+            }).filter(resolver => htmlToUse.startsWith(resolver.getPrefix()))
+                .map(res => {
+                    if(res.isStrict()) {
+                        return res.get(htmlToUse, locale);
+                    }
+
+                    return res.get(htmlToUse, locale.split("-")[0]);
+                })
+                .map(it => reflect(it))
+
+            Promise.all(resolver)
```

Comment:
We want to use the first non-rejected value here. Consider using a "race to success" function where the first successful promise is greedily returned